### PR TITLE
tests(database): added test case for db value guarantee

### PR DESCRIPTION
**Pull request by dackers86**
Android currently does not guarantee that 'child_added' values return before any 'value' listeners.

Please see https://github.com/invertase/react-native-firebase/issues/3940

### CODE REVIEW

File: packages/database/e2e/query/once.e2e.js [hunk 1]
```diff
@@ -181,4 +181,26 @@ describe('database().ref().once()', () => {
     callback.should.be.calledOnce();
     callback.should.be.calledWith({ nuggets: 57 });
   });
+
+  it.only('guarantee to resolve after child_added', async () => {
```
Comment (Salakar) [inline packages/database/e2e/query/once.e2e.js:185]:
```suggestion
  it('guarantee to resolve after child_added', async () => {
```

Comment (dackers86) [reply packages/database/e2e/query/once.e2e.js:185]: I've left this in for now. As this is the one I need to discuss. The implementation is looking pretty tricky!

File: packages/database/e2e/query/once.e2e.js [hunk 2]
```diff
@@ -181,4 +181,26 @@ describe('database().ref().once()', () => {
...
+
+    ref.on('child_added', () => successCallback('child_added'));
+    ref.once('value').then(successCallback('value'));
+
+    await ref.set(initial);
+
+    await Utils.sleep(5000);
+
+    successCallback.should.be.callCount(3);
+    successCallback.getCall(0).args[0].should.equal('child_added');
```
Comment (mikehardy) [inline packages/database/e2e/query/once.e2e.js:202]:
```
2020-07-30T09:32:49.2916850Z 
2020-07-30T09:32:49.2949220Z   1) database().ref().once()
2020-07-30T09:32:49.2949480Z        guarantee to resolve after child_added:
2020-07-30T09:32:49.2949570Z 
2020-07-30T09:32:49.2950800Z       AssertionError: expected 'value' to be 'child_added'
2020-07-30T09:32:49.2951370Z       + expected - actual
2020-07-30T09:32:49.2951460Z 
2020-07-30T09:32:49.2952000Z       -value
2020-07-30T09:32:49.2952190Z       +child_added
2020-07-30T09:32:49.2952350Z       
2020-07-30T09:32:49.2952570Z       at Assertion.fail (node_modules/should/cjs/should.js:275:17)
2020-07-30T09:32:49.2952790Z       at Assertion.value (node_modules/should/cjs/should.js:356:19)
2020-07-30T09:32:49.2953560Z       at Context.it.only (/Users/runner/work/react-native-firebase/react-native-firebase/packages/database/e2e/query/once.e2e.js:202:47)
2020-07-30T09:32:49.2953720Z 
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Appears non-working, not sure what the plan is here? Just curious as I scan through the PR queue

**Comment by Salakar** [comment]
I think the plan was to provide a test case for the linked issue. Happy if you want to close though if this has gone stale

**Comment by dackers86** [comment]
@salakar @mikehardy I'll close this as this can be done through associated PR

**Pull request closed without merging**